cpp/src/arrow/dataset/scanner.h

```diff
@@ -384,6 +381,23 @@ class ARROW_DS_EXPORT ScannerBuilder {
   /// This option provides a control limiting the memory owned by any RecordBatch.
   Status BatchSize(int64_t batch_size);
 
+  /// \brief Set the number of batches to read aheadw ithin a file.
+  ///
+  /// \param[in] batch_readahead How many batches to read ahead within a file,
+  ///  might not work for all formats. refer to comments above.
+  /// \returns An error if this number is less than 0.
+  ///
+  /// This option provides a control on RAM vs IO tradeoff.
```

Comment:
```suggestion
  /// This option provides a control on RAM vs I/O tradeoff.
```

Or maybe something like "higher readahead can improve I/O utilization but will consume more RAM"